src/hooks/useWebData.js

```diff
@@ -81,6 +83,9 @@ export default function useWebData() {
 
   const updateWebProfile = useCallback(
     async (address, name, color) => {
+      const wallet = findWalletWithAccount(address);
+      if (wallet.type === WalletTypes.readOnly) return;
+
       if (!webDataEnabled) return;
```

Comment:
I feel like we could combine both returns

Reply:
yes and no 😅 

I'm gonna move the first return to the top, so we can exit early and maybe prevent the findWalletWithAccount lookup 

🤝